metric/types.go

```diff
@@ -30,3 +35,41 @@ const (
 	TimerType
 	GaugeType
 )
+
+// ValidTypes is a list of valid types.
+var ValidTypes = []Type{
+	CounterType,
+	TimerType,
+	GaugeType,
+}
+
+func (t Type) String() string {
+	switch t {
+	case CounterType:
+		return "counter"
+	case TimerType:
+		return "timer"
+	case GaugeType:
+		return "gauge"
+	default:
+		return "unknown"
+	}
+}
+
+// UnmarshalYAML unmarshals YAML object into a metric type.
+func (t *Type) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var str string
+	if err := unmarshal(&str); err != nil {
+		return err
+	}
+	validTypes := make([]string, 0, len(ValidTypes))
+	for _, valid := range ValidTypes {
+		if str == string(valid) {
```

Comment:
Might want to trim whitespace here.

Reply:
Meh I'd prefer to keep the code simple and don't do sanitization here. This is only for config anyway.